quinn-udp/src/windows.rs

```diff
@@ -40,6 +47,36 @@ impl UdpSocketState {
 
         let sock_true: u32 = 1;
 
+        // Detect if OS expose WSARecvMsg API based on
+        // https://github.com/Azure/mio-uds-windows/blob/a3c97df82018086add96d8821edb4aa85ec1b42b/src/stdnet/ext.rs#L601
+        let guid = WinSock::WSAID_WSARECVMSG;
+        let mut wsa_recvmsg_ptr = None;
+        let mut len = 0;
+
+        // FIXME add Safety: Option handles the NULL pointer with a None value
+        let rc = unsafe {
+            WinSock::WSAIoctl(
+                socket.0.as_raw_socket() as _,
```

Comment:
I see we're passing in the socket here. Does that mean the function pointer might vary depending on the socket? If not, I wonder if we should use a static `OnceLock` or similar to avoid redundant work for every new endpoint.

Reply:
It is the same and can be done on a [dummy socket](https://github.com/microsoft/Windows-classic-samples/blob/main/Samples/Win7Samples/netds/winsock/recvmsg/rmmc.cpp#L477).

I will change the code to use a `OnceLock`.

Reply:
new code uses `OnceCell` from the crate due to rust MSRV, could you take a look and see if it is fine with you ? code will change a bit due to removal of legacy part but principle will stay the same.